From e912b82546052bcac4ff064d18c42336a4d161f6 Mon Sep 17 00:00:00 2001
From: Jason Smith <jsmith@habana.ai>
Date: Sat, 24 May 2025 07:46:37 -0400
Subject: [PATCH] [SW-230673] pldm: extract pdrs for auto-gen

Manual cut-and-paste of the PDRs to be auto-generated.
A future commit will use the script to generate these.

Change-Id: Ie800f9f0bc7ed79f8e28f6be34313c029647ab8e
Signed-off-by: Jason Smith <jsmith@habana.ai>
---

diff --git a/common/include/gaudi3/pldm_pdr.h b/common/include/gaudi3/pldm_pdr.h
index 53cda55..bb247d3 100644
--- a/common/include/gaudi3/pldm_pdr.h
+++ b/common/include/gaudi3/pldm_pdr.h
@@ -117,62 +117,7 @@
 	PLDM_RIDX_EFFECTER_STATE_PCIE_MARGIN_TOOL_MODE_2,
 };
 
-enum pldm_aux_ridx {
-	PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET,
-	PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2,
-	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
-	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
-	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE,
-	PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
-	PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
-	PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
-	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
-	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
-	PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
-	PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
-	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
-	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
-};
-
-enum pldm_entity_aux_ridx {
-	PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD,
-	PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD,
-	PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR,
-};
-
-enum pldm_oem_state_set_ridx {
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE,
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY,
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
-	PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
-};
-
-enum pldm_oem_unit_ridx {
-	PLDM_RIDX_SENSOR_OEM_UNIT_VER_32,
-	PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32,
-	PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16,
-};
-
 extern struct pldm_pdr_record pldm_pdr_records[];
 extern const size_t pldm_pdr_records_length;
-
-extern struct pldm_pdr_name_record pldm_pdr_name_records[];
-extern const size_t pldm_pdr_name_records_length;
-
-extern struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[];
-extern const size_t pldm_pdr_entity_name_records_length;
-
-extern struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[];
-extern const size_t pldm_pdr_oem_state_set_records_length;
-
-extern struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[];
-extern const size_t pldm_pdr_oem_unit_records_length;
 
 #endif /* PLUGINS_RAS_PLDM_PDR_H_ */
diff --git a/common/include/gaudi3/pldm_pdr_generated.h b/common/include/gaudi3/pldm_pdr_generated.h
new file mode 100644
index 0000000..858c5c6
--- /dev/null
+++ b/common/include/gaudi3/pldm_pdr_generated.h
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Habana Labs
+ */
+
+#ifndef PLUGINS_RAS_PLDM_PDR_GENERATED_H_
+#define PLUGINS_RAS_PLDM_PDR_GENERATED_H_
+
+/* NOTE: temporary hand-crafted file; will be auto-generated in the future */
+
+#include "pldm.h"
+#include <pldm_arch_common.h>
+#include <pldm_state_set.h>
+
+enum pldm_aux_ridx {
+	PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET,
+	PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2,
+	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
+	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
+	PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE,
+	PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
+	PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
+	PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
+	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
+	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
+	PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
+	PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
+	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
+	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
+};
+
+enum pldm_entity_aux_ridx {
+	PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR,
+};
+
+enum pldm_oem_state_set_ridx {
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE,
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY,
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
+	PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
+};
+
+enum pldm_oem_unit_ridx {
+	PLDM_RIDX_SENSOR_OEM_UNIT_VER_32,
+	PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32,
+	PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16,
+};
+
+extern struct pldm_pdr_name_record pldm_pdr_name_records[];
+extern const size_t pldm_pdr_name_records_length;
+
+extern struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[];
+extern const size_t pldm_pdr_entity_name_records_length;
+
+extern struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[];
+extern const size_t pldm_pdr_oem_state_set_records_length;
+
+extern struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[];
+extern const size_t pldm_pdr_oem_unit_records_length;
+
+#endif /* PLUGINS_RAS_PLDM_PDR_GENERATED_H_ */
+
diff --git a/common/src/gaudi3/ras/pldm/pldm_effecter_db.c b/common/src/gaudi3/ras/pldm/pldm_effecter_db.c
index 98d8532..071e6a7 100644
--- a/common/src/gaudi3/ras/pldm/pldm_effecter_db.c
+++ b/common/src/gaudi3/ras/pldm/pldm_effecter_db.c
@@ -3,7 +3,9 @@
  * Copyright (c) 2024 Habana Labs
  */
 
-#include "pldm_pdr.h"
+#include <pldm_pdr.h>
+#include <pldm_pdr_generated.h>
+
 #include "pldm_effecter_db.h"
 #include "pldm_effecter_handlers.h"
 #include <pldm_effecter_handlers.h>
diff --git a/common/src/gaudi3/ras/pldm/pldm_pdr.c b/common/src/gaudi3/ras/pldm/pldm_pdr.c
index da82763..f871ed6 100644
--- a/common/src/gaudi3/ras/pldm/pldm_pdr.c
+++ b/common/src/gaudi3/ras/pldm/pldm_pdr.c
@@ -3,12 +3,8 @@
  * Copyright (c) 2024 Habana Labs
  */
 
-/*
- * NOTE: hand-written placeholder file for development.
- *       intended to be replaced by target-specific autogenerated content.
- */
-
 #include <pldm_pdr.h>
+#include <pldm_pdr_generated.h>
 #include <pldm_pcie.h>
 #include <pldm_state_set.h>
 #include <zephyr/sys/util.h>
@@ -884,1236 +880,3 @@
 };
 
 const size_t pldm_pdr_records_length = ARRAY_SIZE(pldm_pdr_records);
-
-/* Variable names are based on the string they represent, with
- * minimal edits to generate a valid C variable name:
- *  - Replace ' ' with '_'
- *  - Remove '.'
- */
-const uint16_t oamAddInCardAccelerator_str[] = {
-	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400, 0x4900,
-	0x6e00, 0x4300, 0x6100, 0x7200, 0x6400, 0x0000
-};
-const uint16_t oamCPLD_str[] = {
-	0x6f00, 0x6100, 0x6d00, 0x4400, 0x5000, 0x4c00, 0x4400,
-	0x0000
-};
-const uint16_t accelerator_str[] = {
-	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500, 0x7200,
-	0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
-};
-const uint16_t asicReset_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5200, 0x6500, 0x7300,
-	0x6500, 0x7400, 0x0000,
-};
-const uint16_t bootStage_str[] = {
-	0x6200, 0x6f00, 0x6f00, 0x7400, 0x5300, 0x7400, 0x6100,
-	0x6700, 0x6500, 0x0000
-};
-const uint16_t fwUpdateState_str[] = {
-	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
-	0x6500, 0x5300, 0x7400, 0x6100, 0x7400, 0x6500, 0x0000
-};
-
-const uint16_t temperature_str[] = {
-	0x7400, 0x6500, 0x6d00, 0x7000, 0x6500, 0x7200, 0x6100,
-	0x7400, 0x7500, 0x7200, 0x6500, 0x0000
-};
-const uint16_t voltage_str[] = {
-	0x7600, 0x6f00, 0x6c00, 0x7400, 0x6100, 0x6700, 0x6500,
-	0x0000
-};
-const uint16_t hbmCATTRIP_str[] = {
-	0x6800, 0x6200, 0x6d00, 0x4300, 0x4100, 0x5400, 0x5400,
-	0x5200, 0x4900, 0x5000, 0x0000
-};
-const uint16_t security_str[] = {
-	0x7300, 0x6500, 0x6300, 0x7500, 0x7200, 0x6900, 0x7400,
-	0x7900, 0x0000
-};
-const uint16_t wd_str[] = {
-	0x7700, 0x6400, 0x0000
-};
-const uint16_t checksum_str[] = {
-	0x6300, 0x6800, 0x6500, 0x6300, 0x6b00, 0x7300, 0x7500,
-	0x6d00, 0x0000
-};
-const uint16_t tempRiseTimeErr_str[] = {
-	0x7400, 0x6500, 0x6d00, 0x7000, 0x5200, 0x6900,
-	0x7300, 0x6500, 0x5400, 0x6900, 0x6d00, 0x6500,
-	0x4500, 0x7200, 0x7200, 0x0000
-};
-const uint16_t fwUpdateError_str[] = {
-	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
-	0x6500, 0x4500, 0x7200, 0x7200, 0x6f00, 0x7200, 0x0000
-};
-const uint16_t revokedPublicKey_str[] = {
-	0x7200, 0x6500, 0x7600, 0x6f00, 0x6b00, 0x6500, 0x6400,
-	0x5000, 0x7500, 0x6200, 0x6c00, 0x6900, 0x6300, 0x4b00,
-	0x6500, 0x7900, 0x0000
-};
-const uint16_t svn_str[] = {
-	0x7300, 0x7600, 0x6e00, 0x0000
-};
-const uint16_t signature_str[] = {
-	0x7300, 0x6900, 0x6700, 0x6e00, 0x6100, 0x7400, 0x7500,
-	0x7200, 0x6500, 0x0000
-};
-const uint16_t key1Revocation_str[] = {
-	0x6b00, 0x6500, 0x7900, 0x3100, 0x5200, 0x6500,
-	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
-	0x6f00, 0x6e00, 0x0000
-};
-const uint16_t key2Revocation_str[] = {
-	0x6b00, 0x6500, 0x7900, 0x3200, 0x5200, 0x6500,
-	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
-	0x6f00, 0x6e00, 0x0000
-};
-const uint16_t key3Revocation_str[] = {
-	0x6b00, 0x6500, 0x7900, 0x3300, 0x5200, 0x6500,
-	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
-	0x6f00, 0x6e00, 0x0000
-};
-const uint16_t key4Revocation_str[] = {
-	0x6b00, 0x6500, 0x7900, 0x3400, 0x5200, 0x6500,
-	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
-	0x6f00, 0x6e00, 0x0000
-};
-const uint16_t key5Revocation_str[] = {
-	0x6b00, 0x6500, 0x7900, 0x3500, 0x5200, 0x6500,
-	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
-	0x6f00, 0x6e00, 0x0000
-};
-const uint16_t fwImageSource_str[] = {
-	0x6600, 0x7700, 0x4900, 0x6d00, 0x6100, 0x6700,
-	0x6500, 0x5300, 0x6f00, 0x7500, 0x7200, 0x6300,
-	0x6500, 0x0000
-};
-const uint16_t memoryCleanup_str[] = {
-	0x6d00, 0x6500, 0x6d00, 0x6f00, 0x7200, 0x7900,
-	0x4300, 0x6c00, 0x6500, 0x6100, 0x6e00, 0x7500,
-	0x7000, 0x0000
-};
-const uint16_t spiWriteProtect_str[] = {
-	0x7300, 0x7000, 0x6900, 0x5700, 0x7200, 0x6900,
-	0x7400, 0x6500, 0x5000, 0x7200, 0x6f00, 0x7400,
-	0x6500, 0x6300, 0x7400, 0x0000
-};
-const uint16_t permanentSecured_str[] = {
-	0x7000, 0x6500, 0x7200, 0x6d00, 0x6100, 0x6e00,
-	0x6500, 0x6e00, 0x7400, 0x5300, 0x6500, 0x6300,
-	0x7500, 0x7200, 0x6500, 0x6400, 0x0000
-};
-const uint16_t Margin_tool_str[] = {
-	0x4d00,  0x6100,  0x7200,  0x6700,  0x6900,  0x6e00,
-	0x2000,  0x7400,  0x6f00,  0x6f00,  0x6c00,  0x0000,
-};
-const uint16_t Eth_Serdes_Availability_str[] = {
-	0x4500,  0x7400,  0x6800,  0x2000,  0x5300,  0x6500,
-	0x7200,  0x6400,  0x6500,  0x7300,  0x2000,  0x4100,
-	0x7600,  0x6100,  0x6900,  0x6c00,  0x6100,  0x6200,
-	0x6900,  0x6c00,  0x6900,  0x7400,  0x7900,  0x0000,
-};
-const uint16_t asicSerialNumberHigh_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
-	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
-	0x6500, 0x7200, 0x4800, 0x6900, 0x6700, 0x6800, 0x0000
-};
-const uint16_t asicSerialNumberLow_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
-	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
-	0x6500, 0x7200, 0x4c00, 0x6f00, 0x7700, 0x0000
-};
-const uint16_t devicePDRRepositoryVersion_str[] = {
-	0x6400, 0x6500, 0x7600, 0x6900, 0x6300, 0x6500, 0x5000,
-	0x4400, 0x5200, 0x5200, 0x6500, 0x7000, 0x6f00, 0x7300,
-	0x6900, 0x7400, 0x6f00, 0x7200, 0x7900, 0x5600, 0x6500,
-	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
-};
-const uint16_t currentPCIeLinkWidth_str[] = {
-	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
-	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5700,
-	0x6900, 0x6400, 0x7400, 0x6800, 0x0000
-};
-const uint16_t currentPCIeLinkSpeed_str[] = {
-	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
-	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5300,
-	0x7000, 0x6500, 0x6500, 0x6400, 0x0000
-};
-const uint16_t pcieFWVersion_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4600, 0x5700, 0x5600, 0x6500,
-	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
-};
-const uint16_t pcieMarginToolModePort0_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3000, 0x0000
-};
-const uint16_t pcieMarginToolModePort1_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x0000
-};
-const uint16_t pcieMarginToolModePort2_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3200, 0x0000
-};
-const uint16_t pcieMarginToolModePort3_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3300, 0x0000
-};
-const uint16_t pcieMarginToolModePort4_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3400, 0x0000
-};
-const uint16_t pcieMarginToolModePort5_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3500, 0x0000
-};
-const uint16_t pcieMarginToolModePort6_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3600, 0x0000
-};
-const uint16_t pcieMarginToolModePort7_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3700, 0x0000
-};
-const uint16_t pcieMarginToolModePort8_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3800, 0x0000
-};
-const uint16_t pcieMarginToolModePort9_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3900, 0x0000
-};
-const uint16_t pcieMarginToolModePort10_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3000,
-	0x0000
-};
-const uint16_t pcieMarginToolModePort11_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3100,
-	0x0000
-};
-const uint16_t pcieMarginToolModePort12_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3200,
-	0x0000
-};
-const uint16_t pcieMarginToolModePort13_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3300,
-	0x0000
-};
-const uint16_t pcieMarginToolModePort14_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3400,
-	0x0000
-};
-const uint16_t pcieMarginToolModePort15_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
-	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
-	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
-	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3500,
-	0x0000
-};
-const uint16_t pcieDeviceID_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4400, 0x6500, 0x7600,
-	0x6900, 0x6300, 0x6500, 0x4900, 0x4400, 0x0000
-};
-const uint16_t pcieVendorID_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x5600, 0x6500, 0x6e00,
-	0x6400, 0x6f00, 0x7200, 0x4900, 0x4400, 0x0000
-};
-
-const uint16_t In_preboot_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
-	0x6F00, 0x6F00, 0x7400, 0x0000
-};
-const uint16_t In_management_FW_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x6E00, 0x6100,
-	0x6700, 0x6500, 0x6D00, 0x6500, 0x6E00, 0x7400, 0x2000,
-	0x4600, 0x5700, 0x0000
-};
-const uint16_t In_preboot_recovery_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
-	0x6F00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x6300,
-	0x6F00, 0x7600, 0x6500, 0x7200, 0x7900, 0x0000
-};
-const uint16_t In_margin_tool_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x7200, 0x6700,
-	0x6900, 0x6E00, 0x2000, 0x7400, 0x6F00, 0x6F00, 0x6C00,
-	0x0000
-};
-const uint16_t In_FW_loader_agent_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x4600, 0x5700, 0x2000, 0x6C00,
-	0x6F00, 0x6100, 0x6400, 0x6500, 0x7200, 0x2000, 0x6100,
-	0x6700, 0x6500, 0x6E00, 0x7400, 0x0000
-};
-const uint16_t Primary_str[] = {
-	0x5000, 0x7200, 0x6900, 0x6D00, 0x6100, 0x7200, 0x7900,
-	0x0000
-};
-const uint16_t Secondary_str[] = {
-	0x5300, 0x6500, 0x6300, 0x6F00, 0x6E00, 0x6400, 0x6100,
-	0x7200, 0x7900, 0x0000
-};
-const uint16_t Not_revoked_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x7600,
-	0x6F00, 0x6B00, 0x6500, 0x6400, 0x0000
-};
-const uint16_t Revoked_str[] = {
-	0x5200, 0x6500, 0x7600, 0x6F00, 0x6B00, 0x6500, 0x6400,
-	0x0000
-};
-const uint16_t Not_permanent_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7000, 0x6500, 0x7200,
-	0x6D00, 0x6100, 0x6E00, 0x6500, 0x6E00, 0x7400, 0x0000
-};
-const uint16_t Permanent_str[] = {
-	0x5000, 0x6500, 0x7200, 0x6D00, 0x6100, 0x6E00, 0x6500,
-	0x6E00, 0x7400, 0x0000
-};
-const uint16_t Not_asserted_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x6100, 0x7300, 0x7300,
-	0x6500, 0x7200, 0x7400, 0x6500, 0x6400, 0x0000
-};
-const uint16_t Asserted_str[] = {
-	0x4100, 0x7300, 0x7300, 0x6500, 0x7200, 0x7400, 0x6500,
-	0x6400, 0x0000
-};
-const uint16_t Only_ber_str[] = {
-	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x4200, 0x4500,
-	0x5200, 0x0000
-};
-const uint16_t Only_diagnostics_str[] = {
-	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x6400, 0x6900,
-	0x6100, 0x6700, 0x6E00, 0x6F00, 0x7300, 0x7400, 0x6900,
-	0x6300, 0x7300, 0x0000
-};
-const uint16_t Ber_and_diagnostics_str[] = {
-	0x4200, 0x4500, 0x5200, 0x2000, 0x6100, 0x6E00, 0x6400,
-	0x2000, 0x6400, 0x6900, 0x6100, 0x6700, 0x6E00, 0x6F00,
-	0x7300, 0x7400, 0x6900, 0x6300, 0x7300, 0x0000
-};
-const uint16_t Skip_str[] = {
-	0x5300, 0x6B00, 0x6900, 0x7000, 0x0000
-};
-const uint16_t Ver32_str[] = {
-	0x5600, 0x6500, 0x7200, 0x3300, 0x3200, 0x0000
-};
-const uint16_t String32_str[] = {
-	0x5300, 0x7400, 0x7200, 0x6900, 0x6e00, 0x6700, 0x3300,
-	0x3200, 0x0000
-};
-const uint16_t Hex16_str[] = {
-	0x4800, 0x6500, 0x7800, 0x3100, 0x3600, 0x0000
-};
-
-struct pldm_pdr_name_record pldm_pdr_name_records[] = {
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_ASIC_RESET,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_RESET,
-		.count = 1,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = asicReset_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_FW_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_FW_STATE,
-		.count = 2,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = bootStage_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = fwUpdateState_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ERROR_STATE_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ERROR_STATE_2,
-		.count = 7,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = temperature_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = voltage_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = tempRiseTimeErr_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = hbmCATTRIP_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = security_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = wd_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = checksum_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_SPI_ERROR_STATE,
-		.count = 3,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = revokedPublicKey_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = svn_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = signature_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_KEY_STATE,
-		.count = 5,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key1Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key2Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key3Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key4Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key5Revocation_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_STATE,
-		.count = 4,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = fwImageSource_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = memoryCleanup_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = spiWriteProtect_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = permanentSecured_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_MARGIN_TOOL,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_MARGIN_TOOL,
-		.count = 1,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Margin_tool_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_OAM,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_1,
-		.count = 6,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_2,
-		.count = 4,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_SERIAL_NUM_HIGH,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = asicSerialNumberHigh_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_SERIAL_NUM_LOW,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = asicSerialNumberLow_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_DEVICE_PDR_REPOSITORY_VERSION,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = devicePDRRepositoryVersion_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_LINK_WIDTH,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = currentPCIeLinkWidth_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_LINK_SPEED,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = currentPCIeLinkSpeed_str,
-		}},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_1,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort0_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort1_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort2_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort3_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort4_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort5_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort6_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort7_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_2,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort8_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort9_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort10_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort11_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort12_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort13_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort14_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort15_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_DEVICE_ID,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name =  pcieDeviceID_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_VENDOR_ID,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = pcieVendorID_str,
-		}},
-	}},
-};
-
-const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
-
-struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] = {
-	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = oamAddInCardAccelerator_str,
-		},
-	}},
-	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_CPLD,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = oamCPLD_str,
-		},
-	}},
-	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = accelerator_str,
-		},
-	}},
-};
-
-const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
-
-struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[] = {
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_FW_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_FW_STATE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_FW_STATE,
-		.count = 5,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_preboot_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_management_FW_str,
-			},
-			{
-				.min_value = 3,
-				.max_value = 3,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_preboot_recovery_str,
-			},
-			{
-				.min_value = 4,
-				.max_value = 4,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_margin_tool_str,
-			},
-			{
-				.min_value = 5,
-				.max_value = 5,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_FW_loader_agent_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Primary_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Secondary_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_KEY,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SECURITY_KEY,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SECURITY_KEY,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_revoked_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Revoked_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_permanent_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Permanent_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_asserted_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Asserted_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
-		.count = 4,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Only_ber_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Only_diagnostics_str,
-			},
-			{
-				.min_value = 3,
-				.max_value = 3,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Ber_and_diagnostics_str,
-			},
-			{
-				.min_value = 4,
-				.max_value = 4,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Skip_str,
-			},
-		},
-	}},
-};
-
-const size_t pldm_pdr_oem_state_set_records_length = ARRAY_SIZE(pldm_pdr_oem_state_set_records);
-
-struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[] = {
-	[PLDM_RIDX_SENSOR_OEM_UNIT_VER_32] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_VER_32,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = Ver32_str,
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_STRING_32,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = String32_str,
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_HEX_16,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = Hex16_str,
-		},
-	}},
-};
-
-const size_t pldm_pdr_oem_unit_records_length = ARRAY_SIZE(pldm_pdr_oem_unit_records);
diff --git a/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c b/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c
new file mode 100644
index 0000000..cdfea96
--- /dev/null
+++ b/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c
@@ -0,0 +1,1240 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright (c) 2024 Habana Labs
+ */
+
+/* NOTE: temporary hand-crafted file; will be auto-generated in the future */
+
+#include <pldm_pdr.h>
+#include <pldm_pdr_generated.h>
+#include <pldm_state_set.h>
+#include <pldm_arch_common.h>
+
+const uint16_t oamAddInCardAccelerator_str[] = {
+	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400, 0x4900,
+	0x6e00, 0x4300, 0x6100, 0x7200, 0x6400, 0x0000
+};
+const uint16_t oamCPLD_str[] = {
+	0x6f00, 0x6100, 0x6d00, 0x4400, 0x5000, 0x4c00, 0x4400,
+	0x0000
+};
+const uint16_t accelerator_str[] = {
+	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500, 0x7200,
+	0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+const uint16_t asicReset_str[] = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5200, 0x6500, 0x7300,
+	0x6500, 0x7400, 0x0000,
+};
+const uint16_t bootStage_str[] = {
+	0x6200, 0x6f00, 0x6f00, 0x7400, 0x5300, 0x7400, 0x6100,
+	0x6700, 0x6500, 0x0000
+};
+const uint16_t fwUpdateState_str[] = {
+	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
+	0x6500, 0x5300, 0x7400, 0x6100, 0x7400, 0x6500, 0x0000
+};
+
+const uint16_t temperature_str[] = {
+	0x7400, 0x6500, 0x6d00, 0x7000, 0x6500, 0x7200, 0x6100,
+	0x7400, 0x7500, 0x7200, 0x6500, 0x0000
+};
+const uint16_t voltage_str[] = {
+	0x7600, 0x6f00, 0x6c00, 0x7400, 0x6100, 0x6700, 0x6500,
+	0x0000
+};
+const uint16_t hbmCATTRIP_str[] = {
+	0x6800, 0x6200, 0x6d00, 0x4300, 0x4100, 0x5400, 0x5400,
+	0x5200, 0x4900, 0x5000, 0x0000
+};
+const uint16_t security_str[] = {
+	0x7300, 0x6500, 0x6300, 0x7500, 0x7200, 0x6900, 0x7400,
+	0x7900, 0x0000
+};
+const uint16_t wd_str[] = {
+	0x7700, 0x6400, 0x0000
+};
+const uint16_t checksum_str[] = {
+	0x6300, 0x6800, 0x6500, 0x6300, 0x6b00, 0x7300, 0x7500,
+	0x6d00, 0x0000
+};
+const uint16_t tempRiseTimeErr_str[] = {
+	0x7400, 0x6500, 0x6d00, 0x7000, 0x5200, 0x6900,
+	0x7300, 0x6500, 0x5400, 0x6900, 0x6d00, 0x6500,
+	0x4500, 0x7200, 0x7200, 0x0000
+};
+const uint16_t fwUpdateError_str[] = {
+	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
+	0x6500, 0x4500, 0x7200, 0x7200, 0x6f00, 0x7200, 0x0000
+};
+const uint16_t revokedPublicKey_str[] = {
+	0x7200, 0x6500, 0x7600, 0x6f00, 0x6b00, 0x6500, 0x6400,
+	0x5000, 0x7500, 0x6200, 0x6c00, 0x6900, 0x6300, 0x4b00,
+	0x6500, 0x7900, 0x0000
+};
+const uint16_t svn_str[] = {
+	0x7300, 0x7600, 0x6e00, 0x0000
+};
+const uint16_t signature_str[] = {
+	0x7300, 0x6900, 0x6700, 0x6e00, 0x6100, 0x7400, 0x7500,
+	0x7200, 0x6500, 0x0000
+};
+const uint16_t key1Revocation_str[] = {
+	0x6b00, 0x6500, 0x7900, 0x3100, 0x5200, 0x6500,
+	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
+	0x6f00, 0x6e00, 0x0000
+};
+const uint16_t key2Revocation_str[] = {
+	0x6b00, 0x6500, 0x7900, 0x3200, 0x5200, 0x6500,
+	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
+	0x6f00, 0x6e00, 0x0000
+};
+const uint16_t key3Revocation_str[] = {
+	0x6b00, 0x6500, 0x7900, 0x3300, 0x5200, 0x6500,
+	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
+	0x6f00, 0x6e00, 0x0000
+};
+const uint16_t key4Revocation_str[] = {
+	0x6b00, 0x6500, 0x7900, 0x3400, 0x5200, 0x6500,
+	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
+	0x6f00, 0x6e00, 0x0000
+};
+const uint16_t key5Revocation_str[] = {
+	0x6b00, 0x6500, 0x7900, 0x3500, 0x5200, 0x6500,
+	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
+	0x6f00, 0x6e00, 0x0000
+};
+const uint16_t fwImageSource_str[] = {
+	0x6600, 0x7700, 0x4900, 0x6d00, 0x6100, 0x6700,
+	0x6500, 0x5300, 0x6f00, 0x7500, 0x7200, 0x6300,
+	0x6500, 0x0000
+};
+const uint16_t memoryCleanup_str[] = {
+	0x6d00, 0x6500, 0x6d00, 0x6f00, 0x7200, 0x7900,
+	0x4300, 0x6c00, 0x6500, 0x6100, 0x6e00, 0x7500,
+	0x7000, 0x0000
+};
+const uint16_t spiWriteProtect_str[] = {
+	0x7300, 0x7000, 0x6900, 0x5700, 0x7200, 0x6900,
+	0x7400, 0x6500, 0x5000, 0x7200, 0x6f00, 0x7400,
+	0x6500, 0x6300, 0x7400, 0x0000
+};
+const uint16_t permanentSecured_str[] = {
+	0x7000, 0x6500, 0x7200, 0x6d00, 0x6100, 0x6e00,
+	0x6500, 0x6e00, 0x7400, 0x5300, 0x6500, 0x6300,
+	0x7500, 0x7200, 0x6500, 0x6400, 0x0000
+};
+const uint16_t Margin_tool_str[] = {
+	0x4d00,  0x6100,  0x7200,  0x6700,  0x6900,  0x6e00,
+	0x2000,  0x7400,  0x6f00,  0x6f00,  0x6c00,  0x0000,
+};
+const uint16_t Eth_Serdes_Availability_str[] = {
+	0x4500,  0x7400,  0x6800,  0x2000,  0x5300,  0x6500,
+	0x7200,  0x6400,  0x6500,  0x7300,  0x2000,  0x4100,
+	0x7600,  0x6100,  0x6900,  0x6c00,  0x6100,  0x6200,
+	0x6900,  0x6c00,  0x6900,  0x7400,  0x7900,  0x0000,
+};
+const uint16_t asicSerialNumberHigh_str[] = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
+	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
+	0x6500, 0x7200, 0x4800, 0x6900, 0x6700, 0x6800, 0x0000
+};
+const uint16_t asicSerialNumberLow_str[] = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
+	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
+	0x6500, 0x7200, 0x4c00, 0x6f00, 0x7700, 0x0000
+};
+const uint16_t devicePDRRepositoryVersion_str[] = {
+	0x6400, 0x6500, 0x7600, 0x6900, 0x6300, 0x6500, 0x5000,
+	0x4400, 0x5200, 0x5200, 0x6500, 0x7000, 0x6f00, 0x7300,
+	0x6900, 0x7400, 0x6f00, 0x7200, 0x7900, 0x5600, 0x6500,
+	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
+};
+const uint16_t currentPCIeLinkWidth_str[] = {
+	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
+	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5700,
+	0x6900, 0x6400, 0x7400, 0x6800, 0x0000
+};
+const uint16_t currentPCIeLinkSpeed_str[] = {
+	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
+	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5300,
+	0x7000, 0x6500, 0x6500, 0x6400, 0x0000
+};
+const uint16_t pcieFWVersion_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4600, 0x5700, 0x5600, 0x6500,
+	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
+};
+const uint16_t pcieMarginToolModePort0_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3000, 0x0000
+};
+const uint16_t pcieMarginToolModePort1_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x0000
+};
+const uint16_t pcieMarginToolModePort2_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3200, 0x0000
+};
+const uint16_t pcieMarginToolModePort3_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3300, 0x0000
+};
+const uint16_t pcieMarginToolModePort4_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3400, 0x0000
+};
+const uint16_t pcieMarginToolModePort5_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3500, 0x0000
+};
+const uint16_t pcieMarginToolModePort6_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3600, 0x0000
+};
+const uint16_t pcieMarginToolModePort7_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3700, 0x0000
+};
+const uint16_t pcieMarginToolModePort8_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3800, 0x0000
+};
+const uint16_t pcieMarginToolModePort9_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3900, 0x0000
+};
+const uint16_t pcieMarginToolModePort10_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3000,
+	0x0000
+};
+const uint16_t pcieMarginToolModePort11_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3100,
+	0x0000
+};
+const uint16_t pcieMarginToolModePort12_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3200,
+	0x0000
+};
+const uint16_t pcieMarginToolModePort13_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3300,
+	0x0000
+};
+const uint16_t pcieMarginToolModePort14_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3400,
+	0x0000
+};
+const uint16_t pcieMarginToolModePort15_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
+	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
+	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
+	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3500,
+	0x0000
+};
+const uint16_t pcieDeviceID_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4400, 0x6500, 0x7600,
+	0x6900, 0x6300, 0x6500, 0x4900, 0x4400, 0x0000
+};
+const uint16_t pcieVendorID_str[] = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x5600, 0x6500, 0x6e00,
+	0x6400, 0x6f00, 0x7200, 0x4900, 0x4400, 0x0000
+};
+
+const uint16_t In_preboot_str[] = {
+	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
+	0x6F00, 0x6F00, 0x7400, 0x0000
+};
+const uint16_t In_management_FW_str[] = {
+	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x6E00, 0x6100,
+	0x6700, 0x6500, 0x6D00, 0x6500, 0x6E00, 0x7400, 0x2000,
+	0x4600, 0x5700, 0x0000
+};
+const uint16_t In_preboot_recovery_str[] = {
+	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
+	0x6F00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x6300,
+	0x6F00, 0x7600, 0x6500, 0x7200, 0x7900, 0x0000
+};
+const uint16_t In_margin_tool_str[] = {
+	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x7200, 0x6700,
+	0x6900, 0x6E00, 0x2000, 0x7400, 0x6F00, 0x6F00, 0x6C00,
+	0x0000
+};
+const uint16_t In_FW_loader_agent_str[] = {
+	0x4900, 0x6E00, 0x2000, 0x4600, 0x5700, 0x2000, 0x6C00,
+	0x6F00, 0x6100, 0x6400, 0x6500, 0x7200, 0x2000, 0x6100,
+	0x6700, 0x6500, 0x6E00, 0x7400, 0x0000
+};
+const uint16_t Primary_str[] = {
+	0x5000, 0x7200, 0x6900, 0x6D00, 0x6100, 0x7200, 0x7900,
+	0x0000
+};
+const uint16_t Secondary_str[] = {
+	0x5300, 0x6500, 0x6300, 0x6F00, 0x6E00, 0x6400, 0x6100,
+	0x7200, 0x7900, 0x0000
+};
+const uint16_t Not_revoked_str[] = {
+	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x7600,
+	0x6F00, 0x6B00, 0x6500, 0x6400, 0x0000
+};
+const uint16_t Revoked_str[] = {
+	0x5200, 0x6500, 0x7600, 0x6F00, 0x6B00, 0x6500, 0x6400,
+	0x0000
+};
+const uint16_t Not_permanent_str[] = {
+	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7000, 0x6500, 0x7200,
+	0x6D00, 0x6100, 0x6E00, 0x6500, 0x6E00, 0x7400, 0x0000
+};
+const uint16_t Permanent_str[] = {
+	0x5000, 0x6500, 0x7200, 0x6D00, 0x6100, 0x6E00, 0x6500,
+	0x6E00, 0x7400, 0x0000
+};
+const uint16_t Not_asserted_str[] = {
+	0x4E00, 0x6F00, 0x7400, 0x2000, 0x6100, 0x7300, 0x7300,
+	0x6500, 0x7200, 0x7400, 0x6500, 0x6400, 0x0000
+};
+const uint16_t Asserted_str[] = {
+	0x4100, 0x7300, 0x7300, 0x6500, 0x7200, 0x7400, 0x6500,
+	0x6400, 0x0000
+};
+const uint16_t Only_ber_str[] = {
+	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x4200, 0x4500,
+	0x5200, 0x0000
+};
+const uint16_t Only_diagnostics_str[] = {
+	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x6400, 0x6900,
+	0x6100, 0x6700, 0x6E00, 0x6F00, 0x7300, 0x7400, 0x6900,
+	0x6300, 0x7300, 0x0000
+};
+const uint16_t Ber_and_diagnostics_str[] = {
+	0x4200, 0x4500, 0x5200, 0x2000, 0x6100, 0x6E00, 0x6400,
+	0x2000, 0x6400, 0x6900, 0x6100, 0x6700, 0x6E00, 0x6F00,
+	0x7300, 0x7400, 0x6900, 0x6300, 0x7300, 0x0000
+};
+const uint16_t Skip_str[] = {
+	0x5300, 0x6B00, 0x6900, 0x7000, 0x0000
+};
+const uint16_t Ver32_str[] = {
+	0x5600, 0x6500, 0x7200, 0x3300, 0x3200, 0x0000
+};
+const uint16_t String32_str[] = {
+	0x5300, 0x7400, 0x7200, 0x6900, 0x6e00, 0x6700, 0x3300,
+	0x3200, 0x0000
+};
+const uint16_t Hex16_str[] = {
+	0x4800, 0x6500, 0x7800, 0x3100, 0x3600, 0x0000
+};
+
+struct pldm_pdr_name_record pldm_pdr_name_records[] = {
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_ASIC_RESET,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ASIC_RESET,
+		.count = 1,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = asicReset_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_FW_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_FW_STATE,
+		.count = 2,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = bootStage_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = fwUpdateState_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ERROR_STATE_2,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ERROR_STATE_2,
+		.count = 7,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = temperature_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = voltage_str,
+			},			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = tempRiseTimeErr_str,
+			},			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = hbmCATTRIP_str,
+			},			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = security_str,
+			},			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = wd_str,
+			},			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = checksum_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_SECURITY_SPI_ERROR_STATE,
+		.count = 3,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = revokedPublicKey_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = svn_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = signature_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_SECURITY_KEY_STATE,
+		.count = 5,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = key1Revocation_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = key2Revocation_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = key3Revocation_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = key4Revocation_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = key5Revocation_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_SECURITY_STATE,
+		.count = 4,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = fwImageSource_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = memoryCleanup_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = spiWriteProtect_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = permanentSecured_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_MARGIN_TOOL,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_MARGIN_TOOL,
+		.count = 1,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Margin_tool_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_OAM,
+		.count = 8,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_1,
+		.count = 6,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_2,
+		.count = 4,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Eth_Serdes_Availability_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ASIC_SERIAL_NUM_HIGH,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = asicSerialNumberHigh_str,
+		}},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_ASIC_SERIAL_NUM_LOW,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = asicSerialNumberLow_str,
+		}},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_DEVICE_PDR_REPOSITORY_VERSION,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = devicePDRRepositoryVersion_str,
+		}},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_LINK_WIDTH,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = currentPCIeLinkWidth_str,
+		}},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_LINK_SPEED,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = currentPCIeLinkSpeed_str,
+		}},
+	}},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_1,
+		.count = 8,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort0_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort1_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort2_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort3_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort4_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort5_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort6_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort7_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_2,
+		.count = 8,
+		.names = {
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort8_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort9_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort10_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort11_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort12_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort13_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort14_str,
+			},
+			{
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = pcieMarginToolModePort15_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_DEVICE_ID,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name =  pcieDeviceID_str,
+		}},
+	}},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID] = { .aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id = PLDM_SID_PCIE_VENDOR_ID,
+		.count = 1,
+		.names = {{
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = pcieVendorID_str,
+		}},
+	}},
+};
+
+const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
+
+struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] = {
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = oamAddInCardAccelerator_str,
+		},
+	}},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_CPLD,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = oamCPLD_str,
+		},
+	}},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = accelerator_str,
+		},
+	}},
+};
+
+const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
+
+struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[] = {
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_FW_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_FW_STATE,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_FW_STATE,
+		.count = 5,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = In_preboot_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = In_management_FW_str,
+			},
+			{
+				.min_value = 3,
+				.max_value = 3,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = In_preboot_recovery_str,
+			},
+			{
+				.min_value = 4,
+				.max_value = 4,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = In_margin_tool_str,
+			},
+			{
+				.min_value = 5,
+				.max_value = 5,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = In_FW_loader_agent_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
+		.count = 2,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Primary_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Secondary_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_KEY,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_SECURITY_KEY,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_SECURITY_KEY,
+		.count = 2,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Not_revoked_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Revoked_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
+		.count = 2,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Not_permanent_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Permanent_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
+		.count = 2,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Not_asserted_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Asserted_str,
+			},
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE] = { .oem_state_set = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.id_handle = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
+		.iana = INTEL_IANA,
+		.id = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
+		.count = 4,
+		.values = {
+			{
+				.min_value = 1,
+				.max_value = 1,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Only_ber_str,
+			},
+			{
+				.min_value = 2,
+				.max_value = 2,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Only_diagnostics_str,
+			},
+			{
+				.min_value = 3,
+				.max_value = 3,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Ber_and_diagnostics_str,
+			},
+			{
+				.min_value = 4,
+				.max_value = 4,
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = Skip_str,
+			},
+		},
+	}},
+};
+
+const size_t pldm_pdr_oem_state_set_records_length = ARRAY_SIZE(pldm_pdr_oem_state_set_records);
+
+struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[] = {
+	[PLDM_RIDX_SENSOR_OEM_UNIT_VER_32] = { .oem_unit = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_VER_32,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_UNIT,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
+		.iana = INTEL_IANA,
+		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = Ver32_str,
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32] = { .oem_unit = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_STRING_32,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_UNIT,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
+		.iana = INTEL_IANA,
+		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = String32_str,
+		},
+	}},
+	[PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16] = { .oem_unit = {
+		.header = {
+			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_HEX_16,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_OEM_UNIT,
+			.record_change_number = 0,
+		},
+		.terminus_handle = 0,
+		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
+		.iana = INTEL_IANA,
+		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = Hex16_str,
+		},
+	}},
+};
+
+const size_t pldm_pdr_oem_unit_records_length = ARRAY_SIZE(pldm_pdr_oem_unit_records);
+
diff --git a/common/src/gaudi3/ras/pldm/sensord_db.c b/common/src/gaudi3/ras/pldm/sensord_db.c
index 00b0b89..7b9179a 100644
--- a/common/src/gaudi3/ras/pldm/sensord_db.c
+++ b/common/src/gaudi3/ras/pldm/sensord_db.c
@@ -4,6 +4,7 @@
  */
 
 #include <pldm_pdr.h>
+#include <pldm_pdr_generated.h>
 
 #include "sensord_db.h"
 #include "sensord_handlers.h"
diff --git a/common/src/ras/pldm/pldm.c b/common/src/ras/pldm/pldm.c
index fc84d2d..4a78335 100644
--- a/common/src/ras/pldm/pldm.c
+++ b/common/src/ras/pldm/pldm.c
@@ -11,6 +11,7 @@
 #include <pldm.h>
 #include <pldm_fu.h>
 #include <pldm_pdr.h>
+#include <pldm_pdr_generated.h>
 #include <common_utils.h>
 #include <pldm_effecter_db.h>
 #include <sensord.h>
